src/CoreFn/Expr.purs

```diff
@@ -107,21 +115,26 @@ data Expr a
   --
   = Literal a (Literal (Expr a))
 
-derive instance eqExpr :: (Generic a, Eq a) => Eq (Expr a)
 derive instance genericExpr :: Generic a => Generic (Expr a)
-derive instance ordExpr :: (Generic a, Ord a) => Ord (Expr a)
+derive instance ordExpr :: Ord a => Ord (Expr a)
 
-instance isForeignExpr :: IsForeign (Expr Unit) where
-  read value = readCoreFnLabel value >>= flip readExpr value
+instance eqExpr :: Eq (Expr a) where
+  eq (Literal _ l1) (Literal _ l2) = l1 == l2
 
-    where
+instance showExpr :: Show (Expr a) where
+  show (Literal _ l) = "(Literal " <> "_" <> " " <> show l <> ")"
```

Comment:
To clarify – while the following type checks, the tests complain because `Foreign` doesn't have `Eq` or `Show` instances.

```purescript
derive instance eqExpr :: Eq a => Eq (Expr a)

instance showExpr :: Show a => Show (Expr a)
  show (Literal x y) = "(Literal " <> show x <> " " <> show y <> ")"
```